### PR TITLE
add filters for asset explorer

### PR DESCRIPTION
**Pull request by nicoecheza**
### Changes

1. **New Filtering System**
- Added a new `Filter` type and filtering functionality
- Introduced `activeFilter` state to track the current filter
- Added a new `Filters` component to display and manage asset filters
- Implemented `mapAssetTypeToFilter` utility function

2. **Tree Structure Refactoring**
- Moved tree-related logic to a separate file (`tree.ts`)
- Introduced new helper functions:
  - `generateAssetTree`: Main function to build the filtered tree structure
  - `getChildren`: Enhanced to handle filtering and searching
  - `getTiles`: New function to get filtered and searched tiles

3. **UI Improvements**
- Simplified the folder view rendering logic
- Removed redundant conditional rendering
- Added filter UI component to the main view

4. **Search Functionality Enhancement**
- Improved search to work alongside filters
- Search now considers both file names and paths
- Better handling of search matches in the tree structure

5. **Code Organization**
- Extracted complex tree manipulation logic into separate utility functions
- Improved type definitions and interfaces
- Better separation of concerns between UI and data management

### 1. Basic Navigation
1. Open the asset explorer
2. Verify the root "scene" folder is visible and selected by default
3. Click through different folders
   - Ensure folders open/close correctly
   - Verify the correct assets are displayed for each folder
   - Check that the navigation path is accurate

### 2. Filter Functionality
1. Check Filter UI
   - Verify the filter buttons are visible
   - Confirm "All" is selected by default
   - Check that all available filters match the asset types in the project

2. Test Each Filter Type
   - Click "3D Models" filter
     * Only .glb/.gltf files should be visible
   - Click "Images" filter
     * Only image files (.png/.jpg/.jpeg) should be visible
   - Click "Sounds" filter
     * Only audio files should be visible
   - Click "All" filter
     * All assets should become visible again

3. Edge Cases
   - Check empty folders when filtered
   - Verify filter state persists when navigating between folders
   - Test filter behavior with no matching assets

### 3. Search Functionality
1. Basic Search
   - Type a file name in the search box
   - Verify matching files appear immediately
   - Check that search is case-insensitive

2. Combined Search and Filters
   - Apply a filter (e.g., "Images")
   - Enter a search term
   - Verify results show only matching files of the filtered type
   - Try different filter + search combinations

### 4. Asset Operations
1. Asset Selection
   - Click different assets
   - Verify correct asset details are shown
   - Check thumbnail previews load correctly

2. Asset Removal
   - Try to remove an asset
   - Verify confirmation modal appears
   - Confirm removal updates the tree correctly
   - Check filter/search results update after removal

### 5. Drag and Drop
1. Drag Operations
   - Verify assets can be dragged
   - Check drag preview appears correctly

### 6. Performance Testing
1. Load Testing
   - Test with a large number of assets
   - Verify filter operations remain responsive
   - Check search performance with many files

### 7. Error Cases
1. Invalid States
   - Test with missing thumbnails
   - Verify behavior with invalid file types

### 8. Visual Verification
1. UI Elements
   - Verify filter buttons are properly styled
   - Check active/hover states of filters
   - Ensure consistent spacing and alignment

### CODE REVIEW

**Review by anicalbano — APPROVED** [review]
✔️ Reviewed changes on editor and approved by QA
All testing steps are working as intended

File: packages/@dcl/inspector/src/components/ProjectAssetExplorer/Filters/Filters.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import { GoClock as RecentIcon } from 'react-icons/go'
+
+import { PropTypes, Filter } from './types'
+
+import './Filters.css'
+
+export function Filters({ filters, active, onClick }: PropTypes) {
+  const getFilter = useCallback((type: Filter) => {
+    switch (type) {
+      case 'all':
```
Comment (cyaiox) [inline packages/@dcl/inspector/src/components/ProjectAssetExplorer/Filters/Filters.tsx:16]: wdy about moving this filter type to an enum?

File: packages/@dcl/inspector/src/components/ProjectAssetExplorer/utils.ts [hunk 2]
```diff
@@ -29,6 +30,8 @@ export function buildAssetTree(paths: string[]): AssetNodeFolder {
               ? 'audio'
               : lowerPath.endsWith('.mp4')
               ? 'video'
+              : lowerPath.endsWith('.jpg') || lowerPath.endsWith('.jpeg') || lowerPath.endsWith('.png')
```
Comment (cyaiox) [inline packages/@dcl/inspector/src/components/ProjectAssetExplorer/utils.ts:33]: wdy about having an array with valid extensions? something like this:

`const validImageExtension = ['.jpg', '.jpeg', '.png']`

```suggestion
              : validImageExtension.some(ext => lowerPath.endsWith(ext))
```
